addons/account/models/account_payment_term.py

```diff
@@ -64,7 +64,11 @@ def _get_amount_due_after_discount(self, total_amount, untaxed_amount):
             if self.early_pay_discount_computation in ('excluded', 'mixed'):
                 discount_amount_currency = self.currency_id.round((total_amount - untaxed_amount) * percentage)
             else:
-                discount_amount_currency = self.currency_id.round(total_amount - (total_amount * (1 - (percentage))))
+                discount_amount_currency = float_round(
```

Comment:
Even the current code can be simplified into:
```suggestion
                discount_amount_currency = self.currency_id.round(total_amount * percentage)
```
I don't understand why you need to put HALF-DOWN on it. Do you have an example plz?

Reply:
I put an example in the commit message: https://github.com/odoo/odoo/pull/152709/commits/8cef52a38e95d3793fe323a10905055ce0267989

Reply:
PR to fix this starting in 16.3: https://github.com/odoo/odoo/pull/163786
Added an additional explanation as well as an extra test.

Because the `HALF-DOWN` rounding method was introduced in 17.0, I went with an alternative approach of rounding it after subtracting it from the total amount.